include/triton/Dialect/TritonNvidiaGPU/Transforms/TMAUtilities.h

```diff
@@ -22,30 +111,44 @@ mlir::LogicalResult createTMADesc(mlir::Value tmaPtr,
 
   auto elemType = op.getBase().getType().getPointeeType();
   auto elemSize = elemType.getIntOrFloatBitWidth() / 8;
+  auto mmaEncoding = llvm::dyn_cast_or_null<gpu::NVMMASharedEncodingAttr>(
+      op.getType().getBlockType().getEncoding());
+  bool fp4Padded = mmaEncoding && mmaEncoding.getFp4Padded();
 
   int32_t contig_dim_size = op.getTensorShape().back();
```

Comment:
I believe this is the size of the packed shape. So using this value as the size of the smem is incorrect. 

To reiterate what I said in https://github.com/triton-lang/triton/pull/6045/files#r1990716965, what we call "dim" or "size" here is interpreted by the HW as the "number of elements of the specified type". So for `BLOCK_K = 128`, even though fp4 elements are packed in the logical shape, the number of elements is still 128.